### DISCUSSION TITLE
Question about scaling base point

### DISCUSSION THREAD

**Discussion by Hairo** [mozman/ezdxf#1065] [Q&A] [answered]
Scaling with the transform module seems to use `(0, 0)` as the base point, which means an object at `(5000, 5000)` scaled with `transform.scale_uniform(obj, 0.5)` will move to `(2500, 2500)`.

Is there any way to change the base point? or scale without moving?

EDIT:
Tried to translate back with:
```
frame_objs = msp.query('*[layer=="Frame"]')
og_position = bbox.extents(frame_objs).extmin

transform.scale_uniform(frame_objs, 0.5)

frame_pos = bbox.extents(frame_objs).extmin
transform.translate(frame_objs, og_point-frame_pos)
```

But now my objects are at `(5000.0000, 4997.5437)` instead of the original position `(5000, 5000)`, seems to be an issue with the bounding box calculation?

**Comment by mozman** [comment, marked as answer]
1. Move the entity from its current position (around the scaling point) to the origin (0, 0, 0) of the coordinate system.
2. Scale the entity while it is positioned at the origin.
3. Move the scaled entity back to its original position around the scaling point.

```python
import ezdxf
from ezdxf.math import Matrix44


def main():
    doc = ezdxf.new()
    msp = doc.modelspace()
    circle = msp.add_circle((100, 100), radius=5)
    doc.saveas("circle_orig.dxf")

    move_to_origin = Matrix44.translate(-100, -100, 0)
    scale = Matrix44.scale(0.5, 0.5, 1)
    move_back = Matrix44.translate(100, 100, 0)
    circle.transform(move_to_origin @ scale @ move_back)

    doc.saveas("circle_scaled_at_center.dxf")


if __name__ == "__main__":
    main()
```

Before the transformation:

![circle_orig](https://github.com/mozman/ezdxf/assets/4230966/2744f290-bff3-44fa-be7f-72301de0a74f)

After the transformation:

![ciclre_scaled_at_center](https://github.com/mozman/ezdxf/assets/4230966/b4d627cf-6e71-40c5-a1c4-eaa084899256)

**Reply by Hairo** [reply]
Thanks that worked, im still getting a slightly different position after translating back but it might be something with my objects, it works with a single object like your example.
